A new option to SHC

manxkim opened the pull request:
This PR implements Ji Hoon Ryoo's spin Hall conductivity using Wannier interpolation, tested and publised(Phys. Rev. B 99, 235113). It is slightly different from Junfeng Qiao's (Phys. Rev. B 98, 214402), which uses some approximations.

To calculate SHC with Ryoo's method, add the parameter "shc_ryoo=.true." in postw90.x input description. It can be used together with other shc parameters by Qiao. Also, it requires .sHu and .sIu files generated from pw2wannier90.x of quantum espresso, implemented by Ryoo and me.

Regards,
Minsu Ghim
Seoul National University

qiaojunfeng committed:
ensure scissors_shift & shc_bandshift produce identical results

qiaojunfeng committed:
remove tab

qiaojunfeng committed:
add test-suite for SPt SHC

qiaojunfeng committed:
add parser

qiaojunfeng committed:
update num_iter

qiaojunfeng committed:
add test-suite for GaAs ac SHC

qiaojunfeng committed:
add test for GaAs ac SHC

qiaojunfeng committed:
remove debugging innfo

qiaojunfeng committed:
print timing info of SHC

qiaojunfeng committed:
fix typo

qiaojunfeng committed:
fix bug when product(berry_kmesh) is divisible by num_nodes

qiaojunfeng committed:
add comments for timing info

qiaojunfeng committed:
move print progress into subroutine

qiaojunfeng committed:
add Ga As rel pseudo for SHC

qiaojunfeng committed:
add ref

qiaojunfeng committed:
fprettified

qiaojunfeng committed:
formatted

qiaojunfeng committed:
add changelog for SHC

qiaojunfeng committed:
use a meaningful var name

qiaojunfeng committed:
rename var name

qiaojunfeng committed:
add Fermi energy check for ac SHC

qiaojunfeng committed:
update comments for SHC

qiaojunfeng committed:
fix typos

qiaojunfeng committed:
add comments for SHC

qiaojunfeng committed:
add comments for SHC

qiaojunfeng committed:
add comments for SHC

qiaojunfeng committed:
update doc for SHC

qiaojunfeng committed:
unify format

qiaojunfeng committed:
add comments for SHC

qiaojunfeng committed:
remove unused vars of berry_get_js_k

qiaojunfeng committed:
remove unused vars of berry_get_js_k

qiaojunfeng committed:
ignore uncompressed files

qiaojunfeng committed:
should ignore uncompressed *.mmn file

qiaojunfeng committed:
add check for Fermi energy when plotting SHC

qiaojunfeng committed:
add check for Fermi energy when plotting SHC

qiaojunfeng committed:
formate the code

qiaojunfeng committed:
update doc for SHC

qiaojunfeng committed:
add space after \version

qiaojunfeng committed:
update \numexamples

qiaojunfeng committed:
add ref for SHC

qiaojunfeng committed:
remove typo

qiaojunfeng committed:
sync with upstream

qiaojunfeng committed:
replace a weird minus sign

qiaojunfeng committed:
decrease berry_kmesh to make tests run faster

qiaojunfeng committed:
increase tolerance for SHC

qiaojunfeng committed:
add pull request number for SHC

qiaojunfeng committed:
update doc for SHC

qiaojunfeng committed:
add 2 kpath & 1 kslice testsuites for SHC

qiaojunfeng committed:
add ignore files for SHC

qiaojunfeng committed:
only use absolute tolerance

qiaojunfeng committed:
fix bugs in python & gnuplot scripts for plotting SHC colored bands

Commit:
change maximum length of file_name

Commit:
extend maximum length of file_name

jryates committed:
Merge pull request #265 from qhapaq-49/dev-swd

[bug fix] Extension maximum length of file_name in kpath and kslice in postw90

qiaojunfeng committed:
sync with develop branch

qiaojunfeng committed:
resolve merge conflicts

qiaojunfeng committed:
use 64bit int in io_wallclocktime

qiaojunfeng committed:
add changelog

jryates committed:
Merge pull request #264 from qiaojunfeng/shc

New feature for postw90: spin Hall conductivity

jryates committed:
Fix typo in warning

VVitale committed:
Fix of small typo in User guide in the definition of slwf_num

jryates committed:
Merge pull request #268 from VVitale/develop

Fix of small typo in the definition of slwf_num in the User guide  (#261)

jryates committed:
Merge branch 'develop' into io_wallclocktime

jryates committed:
Merge pull request #266 from qiaojunfeng/io_wallclocktime

negative return value of io_wallclocktime

jryates committed:
Merge branch 'develop' into develop

jryates committed:
Merge pull request #267 from jryates/develop

Fix typo in warning message

normarivano committed:
Fix issue 260

qiaojunfeng committed:
correct bug in pw90common_fourier_R_to_k_vec when use_ws_distance

qiaojunfeng committed:
fix wrong loop index of wdist_ndeg

qiaojunfeng committed:
revert to old version, will update it later

qiaojunfeng committed:
changelog for pw90common_fourier_R_to_k

qiaojunfeng committed:
add w90spn2spn

qiaojunfeng committed:
skip conjugations on diagonal elements

qiaojunfeng committed:
skip conjugations, ensure identical contents after file conversion

qiaojunfeng committed:
add w90spn2spn.x

jimustafa committed:
add pre-commit hooks

- trailing-whitespace
- end-of-file-fixer

jimustafa committed:
fix trailing whitespace and unnecessary blank lines

jaemolihm committed:
Remove warning for spinor SCDM

jaemolihm committed:
Fix bugs in the SCDM part of pw2wannier90 v6.3: 1) davcio 2) int to double precision

jaemolihm committed:
Fix bugs in the SCDM part of pw2wannier90 v6.4: 1) davcio 2) int to double precision

jaemolihm committed:
Change pwscf/v6.3 compute_amn_with_scdm from using invfft to using slow Fourier transformation to compute psi_nk at pivot points.

jaemolihm committed:
Remove in pwscf/v6.3 one '#if defined(__MPI)' in compute_amn_with_scdm where invfft and gather_grid are no longer called, and mp_sum can be called without #if defined(__MPI).

jaemolihm committed:
Change pwscf/v6.4 compute_amn_with_scdm from using invfft to using slow Fourier transformation to compute psi_nk at pivot points. Also, removed the '#if defined(__MPI)' there.

jaemolihm committed:
Merge branch 'dev_scdm_slowFT' into dev_spinor_scdm

jaemolihm committed:
Implement SCDM for noncollinear wavefunction in pwscf/v6.3/pw2wannier90.f90

jaemolihm committed:
Change mp_pools to mp_global for v6.3/pw2wannier90.f90

jaemolihm committed:
Merge branch 'dev_scdm_slowFT' into dev_spinor_scdm

jaemolihm committed:
Change mp_pools to mp_global for v6.3/pw2wannier90.f90

jaemolihm committed:
Implement SCDM for noncollinear wavefunction in pwscf/v6.4/pw2wannier90.f90

jaemolihm committed:
Changes regarding the comments on PR #277

jaemolihm committed:
Update documentation on the SCDM for spinor wavefunctions: changelog, user guide, tutorial updated, and example31 added.

jaemolihm committed:
Update documentation on the SCDM for spinor wavefunctions: changelog, user guide, tutorial updated, and example31 added.

giovannipizzi committed:
Some minor fixes to PR #277

These change slightly the text of the docs,
fix a compilation problem, and a couple of format
problems in the pw2wannier90 input file for example 31

jaemolihm committed:
Merge pull request #1 from giovannipizzi/fixes_to_pr_277

Some minor fixes to PR #277

jaemolihm committed:
Merge remote branch 'origin/dev_spinor_scdm' into dev_spinor_scdm

Conflicts:
	doc/tutorial/tutorial.tex
	doc/user_guide/projections.tex
	doc/user_guide/wannier-pp.tex
	examples/example31/Pt.pw2wan

jaemolihm committed:
Create pwscf/post-6.4, put the new version of pw2wannier90 from develop branch of Quantum ESPRESSO, and merge the spinor-SCDM-related changes to it.

giovannipizzi committed:
Merge pull request #277 from jaemolihm/dev_spinor_scdm

SCDM for noncollinear systems

giovannipizzi committed:
Adding Lihm as coauthor (for SCDM+spin)

giovannipizzi committed:
Merge pull request #279 from giovannipizzi/add_lihm_coauthor

Adding Lihm as coauthor (for SCDM+spin)

qiaojunfeng committed:
remove unnecessary lines

qiaojunfeng committed:
Merge branch 'develop' into use_ws_distance

qiaojunfeng committed:
use crdist_ws instead of crvec

normarivano committed:
Change#270

jryates committed:
Merge branch 'develop' into fix_issue_260

jryates committed:
Merge pull request #270 from normarivano/fix_issue_260

Fix segmentation fault with `write_xyz`

qiaojunfeng committed:
update code comment

qiaojunfeng committed:
update benchmarks for SHC

qiaojunfeng committed:
update benchmark for wsdistance

jryates committed:
Merge branch 'develop' into use_ws_distance

giovannipizzi committed:
Adding arXiv reference for v3 paper

jryates committed:
Merge branch 'develop' into spn2spn

giovannipizzi committed:
Merge pull request #281 from giovannipizzi/arxiv_reference

Adding arXiv reference for v3 paper

jryates committed:
Merge branch 'develop' into spn2spn

giovannipizzi committed:
Merge branch 'develop' into whitespace-fixes

giovannipizzi committed:
Merge pull request #274 from qiaojunfeng/spn2spn

w90spn2spn.x for conversion between formatted and unformatted spn file

giovannipizzi committed:
Merge branch 'develop' into whitespace-fixes

giovannipizzi committed:
Merge branch 'develop' into use_ws_distance

giovannipizzi committed:
Merge pull request #273 from qiaojunfeng/use_ws_distance

Two bugs in pw90common_fourier_R_to_k*

giovannipizzi committed:
Merge branch 'develop' into whitespace-fixes

giovannipizzi committed:
Adding tests for BoltzWann (at least for the electrical conductivity)

jimustafa committed:
fix trailing whitespace and unnecessary blank lines

jimustafa committed:
prevent trailing whitespace when generating kmesh

giovannipizzi committed:
Gathering matrices before dumping chk file

Fixes #278

In the parallel code, the `u_matrix` is distributed
in `u_matrix_loc`. However, the `write_chkpt` routine
writes on file the content of the `u_matrix` that was not
updated during the wannierisation but only at the end.
Therefore, the code was always restarting from the results
of the disentanglement (unless it was stop appropriately
with a fixed number of steps).

Now we gather the `u_matrix_loc` and `m_matrix_loc` just
before dumping, in the `wannierise.F90` file.
Note that gathering is not needed in the main program file
because the dumping of the .chk file there happens right
after the disentanglement, where the u_matrix is still not
scattered.

Also, I'm not 100% sure that also the `m_matrix_loc` needs to be
gathered, but anyway this is what is done at the end of the
cycle in `wann_main` so it shouldn't harm (at worst it can be
a bit less efficient).

Final note: this is very hard to test automatically, so I
couldn't do a proper testing of this PR.

jryates committed:
Merge pull request #283 from giovannipizzi/fix_278_restart

Gathering matrices before dumping chk file

jryates committed:
Merge branch 'develop' into boltzwann_tests

giovannipizzi committed:
Merge pull request #282 from giovannipizzi/boltzwann_tests

Adding tests for BoltzWann (at least for the electrical conductivity)

giovannipizzi committed:
Merge branch 'develop' into whitespace-fixes

giovannipizzi committed:
Merge pull request #276 from jimustafa/whitespace-fixes

Whitespace fixes

qiaojunfeng committed:
add makefile install target

hjunlee committed:
Add files via upload

hjunlee committed:
Merge pull request #2 from wannier-developers/develop

.

hjunlee committed:
Merge pull request #3 from wannier-developers/develop

.

hjunlee committed:
Add files via upload

hjunlee committed:
Add files via upload

hjunlee committed:
Add files via upload

giovannipizzi committed:
Fix an unused variable that was creating a SEGFAULT

As reported by Nicolas Tancogne-Dejean,
the bnddataunit(n) = io_file_unit() line
was creating a SEGFAULT because the array
was not allocated. But anyway, bnddataunit
is not really used in the code.

jryates committed:
Merge pull request #290 from giovannipizzi/fix_unused_bnddataunit

Fix an unused variable that was creating a SEGFAULT

jryates committed:
Merge branch 'develop' into 190830

jryates committed:
Merge pull request #288 from hjunlee/190830

very minor fix for printing the penalty function

jryates committed:
Merge branch 'develop' into 190901

qiaojunfeng committed:
fix bugs for transforming spn file

qiaojunfeng committed:
fix mixed space and tab

qiaojunfeng committed:
transform to python3 by 2to3

qiaojunfeng committed:
formatted by yapf

qiaojunfeng committed:
fix spelling error

qiaojunfeng committed:
use __file__ to locate kmesh.pl, so k_mapper.py can be executed in any directory

qiaojunfeng committed:
transformed to python 3 syntax

qiaojunfeng committed:
add option write_formatted

qiaojunfeng committed:
add support for non-collinear UNK files & prettify stdout

qiaojunfeng committed:
update documentaion of gw2wannier90.py & k_mapper.py

hjunlee committed:
Merge pull request #4 from wannier-developers/develop

.

hjunlee committed:
Add files via upload

hjunlee committed:
Update berry.F90

Change of the length of the character variable of file_name in order to avoid the truncation.

sponce24 committed:
Update comment for wannier_run called from external code.

sponce24 committed:
Correct spacing issue.

sponce24 committed:
Use pre-commit now.

jryates committed:
Merge pull request #299 from sponce24/develop

Update comment for wannier_run called from external code.

VVitale committed:
Tutorial for extracting SCDM parameters from projectability using projwfc.x

VVitale committed:
Merge branch 'develop' of https://github.com/wannier-developers/wannier90 into develop

VVitale committed:
Tutorial and example for extracting SCDM parameters from projectability

VVitale committed:
Changed text in example32 as suggested by Pizzi

VVitale committed:
typo-fix in citation in example32

VVitale committed:
Further changes to tutorial.tex requested by G. Pizzi

VVitale committed:
Update generate_weights.sh

giovannipizzi committed:
Merge pull request #300 from VVitale/develop

Wannier90 tutorial and example 32 (extracting SCDM parameters from projectability with projwfc.x)

giovannipizzi committed:
Merge branch 'develop' into gw2wannier90

giovannipizzi committed:
Merge pull request #294 from qiaojunfeng/gw2wannier90

bug fixes of gw2wannier90.py & k_mapper.py

qiaojunfeng committed:
add quotes to SCDM parameter in pw2wannier90 input files

giovannipizzi committed:
Updating all places with the references of the new paper

The paper was just published on J. Phys. Cond. Matt.

Since it is Open Access, I am directly removing all
references to the arXiv.

qiaojunfeng committed:
Merge branch 'develop' of github.com:qiaojunfeng/wannier90 into make_install

qiaojunfeng committed:
add DESTDIR variable & doc

giovannipizzi committed:
Merge pull request #306 from qiaojunfeng/fix_scdm_doc

fix pw2wan input files in SCDM example 27

qiaojunfeng committed:
fix intall target

qiaojunfeng committed:
add doc for seedname_tb.dat

qiaojunfeng committed:
add missing if condition on write_tb

giovannipizzi committed:
Merge branch 'develop' into fix_user_guide_tb_dat

giovannipizzi committed:
Merge pull request #309 from qiaojunfeng/fix_write_tb

add missing if condition on write_tb

giovannipizzi committed:
Merge branch 'develop' into fix_user_guide_tb_dat

giovannipizzi committed:
Merge pull request #308 from qiaojunfeng/fix_user_guide_tb_dat

add doc for seedname_tb.dat

giovannipizzi committed:
Merge branch 'develop' into make_install

giovannipizzi committed:
Merge pull request #307 from qiaojunfeng/make_install

add install target for makefile

jryates committed:
Merge branch 'develop' into reference_new_paper

giovannipizzi committed:
Merge pull request #305 from giovannipizzi/reference_new_paper

Updating all places with the references of the new paper

hjunlee committed:
Merge pull request #5 from wannier-developers/develop

.

Commit:
first commit

Commit:
first commit

hjunlee committed:
Merge pull request #6 from hjunlee/master

Master

Commit:
first commit

hjunlee committed:
Merge pull request #7 from hjunlee/master

first commit

giovannipizzi committed:
Merge pull request #289 from hjunlee/190901

small restructuring of the routines for the AHC calculation with multiple Fermi energies and adaptive refinement

giovannipizzi committed:
Merge branch 'develop' into 191015

giovannipizzi committed:
Merge pull request #295 from hjunlee/191015

Minor fix and modification of compute_orb in pw2wannier90.f90

giovannipizzi committed:
Moving the new pw2wannier90 file to folder v6.5

Since now v6.5 of Quantum ESPRESSO has been released

giovannipizzi committed:
Preparing release v3.1 of Wannier90

giovannipizzi committed:
Merge pull request #310 from giovannipizzi/move_pw2wan_v65

Moving the new pw2wannier90 file to folder v6.5

giovannipizzi committed:
Merge pull request #311 from giovannipizzi/release_3.1

Preparing release v3.1 of Wannier90

VVitale committed:
Bugfix in disentanglement + symmetry adapted WFs

VVitale committed:
Merge branch 'develop' of https://github.com/wannier-developers/wannier90 into develop

VVitale committed:
Added test for sawfs with disentanglement

VVitale committed:
Update jobconfig

VVitale committed:
Bugfix in testw90_disentanglement_sawfs

VVitale committed:
Merge branch 'develop' of https://github.com/VVitale/wannier90 into develop

VVitale committed:
Reduced threshold to userconfig and jobconfig for testw90_disentanglement_sawfs

VVitale committed:
Implementation of bugfix discovered by Jonathan Yates for disentanglement+sawfs

jryates committed:
Merge pull request #315 from VVitale/develop

Fix a bug in the disentanglement step with SAWFs

giovannipizzi committed:
Release 3.1 (fixed final date)

giovannipizzi committed:
Merge pull request #320 from giovannipizzi/release_3.1

Release 3.1

giovannipizzi committed:
Merge pull request #321 from giovannipizzi/master_3.1_in_develop

Merging back master branch in develop after v3.1 release

qiaojunfeng committed:
increase SHC tests tolerance

giovannipizzi committed:
Merge pull request #323 from qiaojunfeng/fix_shc_tests

increase SHC tests tolerance

giovannipizzi committed:
Adding units of derivative of bands in geninterp

giovannipizzi committed:
Pointless commit to trigger tests

giovannipizzi committed:
Merge pull request #325 from giovannipizzi/fix_units_derivative_geninterp

Adding units of derivative of bands in geninterp

Commit:
corrected the gyrotropic section manual

Commit:
added checks befor bcast for gyrotropic arrays

giovannipizzi committed:
Merge pull request #330 from stepan-tsirkin/fix-doc-gyrotropic

corrected the gyrotropic section manual

Commit:
timing stop was missing in the first call of hamiltonian_wigner_seitz (count_pts=.true.)

Commit:
Updated wigner_seitz subroutine in postw90 to work with ws_search_size as in wannier90

giovannipizzi committed:
Merge pull request #332 from gibertini/develop

Adding ws_search_size also in postw90

Commit:
fix the issue in plot_bvec

giovannipizzi committed:
Merge branch 'develop' into fix-gyrotropic-hard-checks

giovannipizzi committed:
Merge pull request #335 from stepan-tsirkin/fix-gyrotropic-hard-checks

Broadcast variables for gyrotropic only when arrays are defined

hjunlee committed:
Merge pull request #1 from wannier-developers/develop

recent updates in W90

giovannipizzi committed:
Merge pull request #337 from hjunlee/develop

fix the issue in plot_bvec

hjunlee committed:
Update plot.F90

mostofi committed:
Merge pull request #340 from hjunlee/develop

Very minor issue in the subroutine of internal_cube_format in plot.F90

manxkim committed:
Add files via upload

Added Ryoo's option to SHC (PRB 99.235113) using SAA, SBB from get_oper.F90.

qiaojunfeng committed:
fix typo in wannier_plot_spinor_mode parameter

jryates committed:
Merge pull request #345 from qiaojunfeng/fix_plot_doc

fix typo in documentation of wannier_plot_spinor_mode parameter

jimustafa committed:
implement GitHub Actions for `pre-commit` checks

- add the `pre-commit` action
- remove obsolete environments from `.travis.yml`
- remove obsolete scripts from `test-suite`

manxkim committed:
Merge remote-tracking branch 'origin/SHC' into SHC

manxkim committed:
add Ryoo's SHC option

manxkim committed:
add Ryoo's SHC option

manxkim committed:
delete lines for testing

jryates commented:
Thanks for this.

I haven't read the newer paper, or thought about the physics yet. But a few comments:

1- I think it would be more in keeping with W90 style to have a parameter shc_method with values qiao or rhoo (or perhaps something more physical?) rather than shc_ryoo=.true. [perhaps @giovannipizzi or @mostofi can comment]

2- Can you add a test case?

3- Can you update the user guide (after the keyword is decided)

4- There are a few debug lines commented out. It is cleaner for the code base if these are removed. We do ignore this sometime if there is good reason to leave in uncommented code - so use your judgement.

5- The pre-commit checks failed. You need to install the pre-commit hooks, see
https://github.com/wannier-developers/wannier90/wiki/ContributorsGuide

giovannipizzi commented:
Hi all!
1. I agree, it would be better to have something like `shc_method` or `shc_implementation`
2. Indeed, is there an easy way to summarise the difference between the two methods, ideally in one word? :-) Otherwise I guess we're left to the first-author's name
3. I suggest that @qiaojunfeng helps reviewing this PR, as he's the author of the "other" method :-)
4. about the comment `it requires .sHu and .sIu files generated from pw2wannier90.x of quantum espresso, implemented by Ryoo and me`, did you already commit this into the [`pw2wannier90.x` of the Quantum ESPRESSO code](https://gitlab.com/QEF/q-e/-/blob/develop/PP/src/pw2wannier90.f90)? If not, would it be possible to make this available before we merge this PR? Ideally committing directly there, but we also keep a copy of the "patched" versions here, in the pwscf folder (i.e. the file that needs to be replaced for the most recent QE release to work - 6.7 at the moment).

manxkim committed:
add a reference to the new option of SHC

jimustafa committed:
migrate to GitHub Actions

- move `pre-commit` workflow to `main.yml`
- create new build and test workflow
- move all test configuration and scripts to `workflows` directory
- remove obsolete `travis` files

jimustafa committed:
update `pre-commit` config

- normalize to a top-level map
- change `sha` key to `rev`
- remove obsolete `fprettify` wrapper

giovannipizzi committed:
Merge pull request #349 from jimustafa/develop

implement GitHub Actions for all tests, replacing Travis

manxkim committed:
a change in the name of a param. : shc_ryoo to shc_method

manxkim committed:
Removed debug lines commented out

manxkim committed:
change in coding style and etc

manxkim committed:
update the user guide

adding a new option to spin Hall conductivity

manxkim committed:
add a test case to test the new option to SHC

manxkim committed:
reduce file sizes

manxkim committed:
Merge branch 'develop' of github.com:wannier-developers/wannier90 into SHC

manxkim committed:
Merge branch 'SHC' of github.com:manxkim/wannier90 into SHC

manxkim committed:
Merge branch 'SHC' of github.com:manxkim/wannier90 into SHC

manxkim committed:
Merge remote-tracking branch 'upstream/develop' into SHC

manxkim committed:
resolve errors in tests files

manxkim committed:
resolve some checks problems; add files unintendedly deleted

manxkim commented:
Thank you all for comments,

Because I made too many mistakes of pushes and commits("This branch is 740 commits ahead of wannier-developers:develop."), I decided to move to a new branch and pull request newly. My apologies for that, please see https://github.com/wannier-developers/wannier90/pull/353
